src/map-marker/badge-enhancer.js

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright (c) Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+
+// @flow
+import * as React from 'react';
+import type {BadgeEnhancerComponentT} from './types.js';
+import {useStyletron} from '../styles/index.js';
+import {getOverrides} from '../helpers/overrides.js';
+
+import {BadgeEnhancer as StyledBadgeEnhancer} from './styled-components.js';
+import {
+  PINHEAD_TYPES,
+  BADGE_ENHANCER_SIZES,
+  BADGE_ENHANCER_POSITIONS,
+  BADGE_ENHANCER_CONTENT_SIZE,
+} from './constants.js';
+
+const BadgeEnhancer = ({
+  pinHeadSize,
+  markerType,
+  badgeEnhancerSize = BADGE_ENHANCER_SIZES.none,
+  badgeEnhancerColor,
+  badgeEnhancerBackground,
+  badgeEnhancerContent: BadgeEnhancerContent,
+  overrides = {},
+}: BadgeEnhancerComponentT) => {
+  const [, theme] = useStyletron();
+  const {
+    colors: {backgroundAccent, primaryB},
+  } = theme;
+
+  if (
+    badgeEnhancerSize === null ||
+    badgeEnhancerSize == BADGE_ENHANCER_SIZES.none
+  ) {
+    return null;
+  }
+  if (
+    badgeEnhancerSize !== BADGE_ENHANCER_SIZES.xSmall &&
+    !BadgeEnhancerContent
+  ) {
+    console.warn(
+      `Badges (except for size ${BADGE_ENHANCER_SIZES.xSmall}) must contain content`,
+    );
+    return null;
+  }
```

Comment:
We usually add these warnings within a DEV fence so that it can be compiled out of production builds. Same with others below here
```suggestion
  if (__DEV__) {
    if (
      badgeEnhancerSize !== BADGE_ENHANCER_SIZES.xSmall &&
      !BadgeEnhancerContent
    ) {
      console.warn(
        `Badges (except for size ${BADGE_ENHANCER_SIZES.xSmall}) must contain content`,
      );
      return null;
    }
  }
```